src/Symfony/Component/Yaml/Inline.php

```diff
@@ -270,97 +261,95 @@ private static function dumpArray($value, $flags)
         return sprintf('{ %s }', implode(', ', $output));
     }
 
+    private static function parseValue(StringReader $reader, $flags = 0, $delimiters = null, $evaluate = true, $references = array())
+    {
+        try {
+            return $reader->transact(function () use ($reader, $flags, $delimiters, $evaluate, $references) {
+                $reader->eatWhiteSpace();
+                if ($reader->eat('[')) {
+                    return self::parseSequence($reader, $flags, $references);
+                }
+                if ($reader->eat('{')) {
+                    return self::parseMapping($reader, $flags, $references);
+                }
+
+                $scalar = self::parseScalar($reader, $flags, $delimiters, $evaluate, $references);
+
+                return $scalar;
+            });
+        } catch (UnderflowException $e) {
+            throw new ParseException(sprintf('Malformed inline YAML string %s.', $reader->readToFullConsumption()));
+        } catch (ParseException $e) {
+            throw new ParseException(sprintf('Malformed inline YAML string %s. %s', $reader->readToFullConsumption(), $e->getMessage()));
+        }
+    }
+
     /**
      * Parses a scalar to a YAML string.
      *
-     * @param string $scalar
-     * @param int    $flags
-     * @param string $delimiters
-     * @param array  $stringDelimiters
-     * @param int    &$i
-     * @param bool   $evaluate
-     * @param array  $references
+     * @param StringReader $reader
+     * @param int          $flags
+     * @param string       $delimiters
+     * @param bool         $evaluate
+     * @param array        $references
      *
      * @return string A YAML string
      *
      * @throws ParseException When malformed inline YAML string is parsed
-     *
-     * @internal
```

Comment:
@xabbuh if someone looks at the `3.1` code he will see that this method is `@internal` because its class is. What's the problem?